http-native/src/main/java/org/ballerinalang/net/transport/contractimpl/sender/states/http2/ReceivingHeaders.java

```diff
@@ -203,6 +207,29 @@ private void onResponseHeadersRead(ChannelHandlerContext ctx, OutboundMsgHolder
         }
     }
 
+    private void checkForTrailerHeaders(int streamId, Http2Headers http2Headers, OutboundMsgHolder outboundMsgHolder,
```

Comment:
As per my understanding the responsibility of this method is to process the request if the trailer-header exist. So, better to name it as something like `processHttp2Trailer` etc.

Reply:
As I have understood, the method first checks whether there is a trailer-header specified and not. And if trailer is not specified it complete the response with default last http content. Even there is a trailer specified, the method checks whether the respective trailers exists or not and if not it again complete the response with default last content. So the method does not simply process trailer, it checks whether trailer is present and call corresponding methods.